src/components/Layout/Header/NavBar/Notifications.tsx

```diff
@@ -40,14 +62,47 @@ export const Notifications = () => {
     }
   }, [colorMode, mobileBreakpoint])
 
-  if (!isWhereverEnabled || !currentWallet || !isKeyManagerWithProvider(currentWallet)) return null
+  if (
+    !isWhereverEnabled ||
+    !currentWallet ||
+    !eip712SupportedWallets.includes(currentWallet) ||
+    !wallet ||
+    !supportsETH(wallet)
+  )
+    return null
 
-  const disableAnalytics = window.location.href.includes('private.shapeshift.com')
-  const partnerKey = getConfig().REACT_APP_WHEREVER_PARTNER_KEY
+  const { addressNList } = wallet.ethGetAccountPaths({
+    coin: 'Ethereum',
+    accountIdx: 0,
+  })[0]
+
+  const signMessage = async (message: string) => {
+    const signedMsg = await wallet.ethSignMessage({
+      addressNList,
+      message,
+    })
+
+    return signedMsg?.signature
+  }
+
+  const signTypedData = async (typedData: ETHSignTypedData['typedData']) => {
+    const signedMsg = await wallet.ethSignTypedData?.({
+      addressNList,
+      typedData,
+    })
+
+    return signedMsg?.signature
+  }
 
   return (
     <Box>
       <NotificationFeedProvider
+        customSigner={{
+          address: ethAddress as string,
```

Comment:
we shouldn't have to cast here, should be able to correctly infer a `string` above

Reply:
As this is now part of state, left as is